xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/changesdoc.vm

```diff
@@ -24,40 +24,44 @@
 
 #if ("$!previousOrigdocVersion" != '')
   #set ($changesLink = ${xwiki.getURL($tdoc, 'view', "viewer=changes&rev1=${previousOrigdocVersion}&rev2=${rev2}")})
-  #set ($changesTitle = $services.localization.render('core.viewers.diff.previousVersion'))
-  #set ($previousOrigdocLink = "<a class='changes-arrow' href='$changesLink' title='$changesTitle'>&lt;</a>")
+  #set ($changesTitle = $escapetool.html($services.localization.render('core.viewers.diff.previousVersion.origin', [${previousOrigdocVersion}])))
+  #set ($previousOrigdocLink = "<a class='btn btn-default changes-arrow' href='$changesLink' title='$changesTitle'>$services.icon.renderHTML('left')</a>")
 #end
 #if ("$!nextOrigdocVersion" != '')
   #set ($changesLink = ${xwiki.getURL($tdoc, 'view', "viewer=changes&rev1=${nextOrigdocVersion}&rev2=${rev2}")})
-  #set ($changesTitle = $services.localization.render('core.viewers.diff.nextVersion'))
-  #set ($nextOrigdocLink = "<a class='changes-arrow' href='$changesLink' title='$changesTitle'>&gt;</a>")
+  #set ($changesTitle = $escapetool.html($services.localization.render('core.viewers.diff.nextVersion.origin', [${nextOrigdocVersion}])))
+  #set ($nextOrigdocLink = "<a class='btn btn-default changes-arrow' href='$changesLink' title='$changesTitle'>$services.icon.renderHTML('right')</a>")
 #end
 #if ("$!previousNewdocVersion" != '')
   #set ($changesLink = ${xwiki.getURL($tdoc, 'view', "viewer=changes&rev1=${rev1}&rev2=${previousNewdocVersion}")})
-  #set ($changesTitle = $services.localization.render('core.viewers.diff.previousVersion'))
-  #set ($previousNewdocLink = "<a class='changes-arrow' href='$changesLink' title='$changesTitle'>&lt;</a>")
+  #set ($changesTitle = $escapetool.html($services.localization.render('core.viewers.diff.previousVersion.target', [${previousNewdocVersion}])))
```

Comment:
I think it should be fine here, but `$escapetool.html` doesn't escape `{` which might be dangerous if this template is included in a Velocity macro inside a nested HTML macro as it doesn't prevent closing the HTML macro. That's why we normally use `$escapetool.xml`.

Reply:
:+1: Addressed in https://github.com/xwiki/xwiki-platform/pull/2323/commits/082bbf5d74b6d34d5c99a4e90fe1eebca9a3bc28